pkg/trustedresources/verify.go

```diff
@@ -33,20 +33,40 @@ import (
 	"github.com/tektoncd/pipeline/pkg/trustedresources/verifier"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/apis"
 	"knative.dev/pkg/logging"
 )
 
 const (
 	// SignatureAnnotation is the key of signature in annotation map
 	SignatureAnnotation = "tekton.dev/signature"
+	// ConditionTrustedResourcesVerified specifies that the resources pass trusted resources verification or not.
+	ConditionTrustedResourcesVerified apis.ConditionType = "TrustedResourcesVerified"
+	IgnoreNoMatchPolicy                                  = iota
+	WarnNoMatchPolicy
+	VerificationPass
+	WarnModeVerificationFail
 )
 
+type VerificationResultType int
+
+// VerificationResult contains the type and message of cases when the verification doesn't return error.
+// The VerificationResultType has 4 types which is corresponding to 4 cases:
+// 1) IgnoreNoMatchPolicy: The verification is skipped when there's no matching policies.
+// 2) WarnNoMatchPolicy: There's no matching policies and warning is logged.
+// 3) VerificationPass: The verification passes.
+// 4) WarnModeVerificationFail: Only Warn mode verification policies fail.
+type VerificationResult struct {
+	VerificationResultType VerificationResultType
+	Message                string
+}
+
 // VerifyTask verifies the signature and public key against task.
 // Skip the verification when no policies are found and trusted-resources-verification-no-match-policy is set to ignore or warn
 // Return an error when no policies are found and trusted-resources-verification-no-match-policy is set to fail,
 // or the resource fails to pass matched enforce verification policy
 // refSource contains the source information of the task.
-func VerifyTask(ctx context.Context, taskObj *v1beta1.Task, k8s kubernetes.Interface, refSource *v1beta1.RefSource, verificationpolicies []*v1alpha1.VerificationPolicy) error {
+func VerifyTask(ctx context.Context, taskObj *v1beta1.Task, k8s kubernetes.Interface, refSource *v1beta1.RefSource, verificationpolicies []*v1alpha1.VerificationPolicy) (*VerificationResult, error) {
```

Comment:
VerifyTask and VerifyPipeline share lots of common code, I think we may merge them into 1 function in a follow-up pr